### ISSUE TITLE
Any solutions to integrate Formly with mat-table

### ISSUE THREAD

**Issue by freon27** [ngx-formly/ngx-formly#2534]
I think I probably already know that this is a non-starter since I can't really find any info going through your GH issues, but I'm looking at ways to integrate Formly into mat-tables.

As I see if, both mat-table and Formly want to be in charge of rendering items as a single unit, and I can't see a way to make the two play nicely.... but I'm not an expert with Formly so I thought I would ask to make sure I'm not missing something.

Am I right in my understanding that there is no way of rendering fragments of a Formly form separately?

Thanks in advance

**Comment by aitboudad** [comment]
Actually, there are two approaches for data table:

1. Through a custom type https://formly.dev/examples/advanced/datatable-integration, but not suitable when dealing with large data at once.
2. Inline edit which you can open `formly-form` on demand (double-click for inline edit or inside a popup), so here just `formly-form` would suffice

**Comment by alindberg** [comment]
I am having the same issue.  I need a material table `<mat-table>` supporting a `<formly-field>` to allow for inline editing.  Using a pop-up for input is much to distracting given the number of rows that must be entered.

I created an example on [stackblitz](https://stackblitz.com/edit/angular-material-formly-table). 

The `add()` and `remove()` fucntions defined in `field-array-type`, while they add/remove rows from the `model` and `form.value`  objects, the don't update the `field` object.  This then keeps the table from adding new rows.

To try to fix this, I created my own `add()` and `remove()` functions.  They to force the `dataSouce`, then call `super.add` or `super.remove` then call `this.table.renderRows()`.

The end result is that the everything seems to work on `add`, however on `remove()` the model and `form.values` are adjusted as expected, but the `field.fieldGroup` structure is not.  The function `this.form.removeControl()` will remove the control group but it does not shrink the form.  After a `remove` bad things happen.

Any suggestions on how to resolve this would be great
